src/Hl7.Fhir.Support.Poco/Utility/FhirVersionUtility.cs

```diff
@@ -0,0 +1,112 @@
+﻿using Hl7.Fhir.Specification;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Hl7.Fhir.Utility
+{
+    public class FhirReleaseParser
+    {
+        /// <summary>
+        /// Returns a FHIR version as an enum from a version number
+        /// </summary>
+        /// <param name="version">Fhir Release version number</param>
+        /// <returns>Official FHIR Release</returns>
+        public static FhirRelease Parse(string version)
+        {
+            return version switch
```

Comment:
I think this switch is too specific for "final release versions".  Take 4.0.x -> we know that this is R4, so to avoid us having to directly update the library for every single patch version.

My code looked like this:

``` csharp
 public static FhirRelease DeriveReleaseFromVersion(string fhirVersion)
        {
            if (fhirVersion.StartsWith("5.0")) return FhirRelease.R5;
            if (fhirVersion is "4.2.0" or "4.4.0" or "4.5.0") return FhirRelease.R5;
            if (fhirVersion.StartsWith("4.0")) return FhirRelease.R4;
            if (fhirVersion is "3.6.0" or "3.5a.0" or "3.5.0" or "3.4.0" or "3.3.0" or "3.2.0" or "3.1.0") return FhirRelease.R4;
            if (fhirVersion.StartsWith("3.0")) return FhirRelease.STU3;
            if (fhirVersion is "1.9.0" or "1.8.0" or "1.6.0" or "1.2.0" or "1.1.0") return FhirRelease.STU3;
            if (fhirVersion.StartsWith("1.0")) return FhirRelease.DSTU2;
            if (fhirVersion is "0.5.0" or "0.4.0") return FhirRelease.DSTU2;

            throw new NotSupportedException($"This software does not support FHIR version {fhirVersion}.");
        }
```

As you can see I switch to `StartsWith` when we know it's a final release.  Also, I noticed your code has a few more cases (especially for DSTU1), but this code also has a few more in comparison to yours.  Whatever we chose to do, we need to be as complete as possible!